core/arch/arm/tee/entry_fast.c

```diff
@@ -242,4 +242,7 @@ void __weak tee_entry_get_os_revision(struct thread_smc_args *args)
 {
 	args->a0 = CFG_OPTEE_REVISION_MAJOR;
 	args->a1 = CFG_OPTEE_REVISION_MINOR;
+#ifdef CFG_OS_REV_REPORTS_GIT_SHA1
+	args->a2 = TEE_IMPL_GIT_SHA1;
+#endif
```

Comment:
Wouldn't it be more friendly to assign zero in case this feature isn't enabled, instead of leaving it unchanged?

Reply:
If so, maybe use `CFG_OS_REV_REPORTS_GIT_SHA1` when generating the SHA1 (fill it with 0's).
Nice though @jenswi-linaro about product considerations.

Reply:
Well, the caller can initialize it to zero or whatever special value to mean "unspecified", and actually it has to if it wants to be compatible with the previous implementation. So I'm not sure it is really needed.

Reply:
Yes, the caller can do that. But is that more friendly?

Reply:
OK. Then I think I should also update "core: arm: GET_OS_REVISION: document a2 as a build identifier" and write that zero has a special meaning: "not specified", OK?

Reply:
Sounds good